pkg/apis/serving/v1beta1/trained_model.go

```diff
@@ -0,0 +1,48 @@
+package v1beta1
+
+import (
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	duckv1beta1 "knative.dev/pkg/apis/duck/v1beta1"
+)
+
+type TrainedModel struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+	Spec              TrainedModelSpec         `json:"spec,omitempty"`
+	Status            TrainedModelDeployStatus `json:"status,omitempty"`
+}
+type TrainedModelSpec struct {
+	// Required field for parent inference service
+	InferenceService string `json:"inferenceService"`
+	// Predictor model spec
+	PredictorModel ModelSpec `json:"predictorModel"`
+	// Explainer model spec
+	ExplainerModel ModelSpec `json:"explainerModel,omitempty"`
+}
+type ModelSpec struct {
+	// Storage URI for the model repository
+	StorageURI string `json:"storageUri"`
+	// Default to latest
+	ModelVersionPolicy ModelVersionPolicy `json:"modelVersionPolicy,omitempty"`
+	// ML framework name
+	// +kubebuilder:webhooks:Enum={"tensorflow","pytorch","sklearn","onnx","xgboost", "custom"}
+	Framework string `json:"framework"`
+	// Framework version for the trained model
+	FrameworkVersion string `json:"frameworkVersion,omitempty"`
+	// Defaults to requests and limits of 1CPU, 2Gb MEM.
+	Resources v1.ResourceRequirements `json:"resources,omitempty"`
+}
+type ModelVersionPolicy struct {
```

Comment:
I'm not convinced on this yet. Do we have use cases for users serving multiple model versions in the same inference service? "All" seems kind of dangerous. It also makes "Status" more complex. Would you be willing to punt on versioning for now and we can add it as we understand this more?

Reply:
I am trying to be consistent with triton server's version policy here: https://docs.nvidia.com/deeplearning/triton-inference-server/master-user-guide/docs/model_configuration.html?highlight=policy#version-policy. Triton provides all, specific and latest options. Tf-serving also allows users to server multiple versions at the same time: https://www.tensorflow.org/tfx/serving/serving_config

I do agree that it will make the status object more complicated and I am willing to start with the "latest" version and add support for "specific" and "all" as we gain more experience.

Reply:
Ready to approve as soon as you remove this :)